chainer/training/extensions/_snapshot.py

```diff
@@ -213,6 +293,36 @@ def __init__(
         self.condition = condition
         self.writer = writer
         self._snapshot_on_error = snapshot_on_error
+        self.num_retain = num_retain
+        self.autoload = autoload
+
+    def initialize(self, trainer):
+        if self.autoload:
+            target = trainer if self._target is None else self._target
+
+            outdir = trainer.out
+            filename = find_latest_snapshot(self.filename, outdir)
+            if filename is None:
+                if chainer.is_debug():
+                    print("No snapshot file that matches {} was found"
+                          .format(self.filename))
+            else:
+                snapshot_file = os.path.join(outdir, filename)
+                npz.load_npz(snapshot_file, target)
+                if chainer.is_debug():
+                    print("Snapshot loaded from", snapshot_file)
+
+        if hasattr(self.writer, '_add_cleanup_hook') and \
+           self.num_retain > 0 and isinstance(self.filename, str):
+
+            def _cleanup(filename, outdir, target, savefun, **kwds):
```

Comment:
This is no longer exposed to users. Please remove unnecessary arguments.
(it looks no argument is needed?)